File: spmercantil/src/main/java/github/faustofjunqueira/spmercantil/application/adapters/controller/response/PageResponse.java
package github.faustofjunqueira.spmercantil.application.adapters.controller.response;

import lombok.Data;

import java.io.Serializable;
import java.util.List;

/**
 * Representação de registros paginado da base de dados
 *
 * @param <TData> Tipo do registro carregado
 */
@Data
public class PageResponse<TData> {

    /**
     * total de registro presentes na base de dados
     */
    private Long total;

    /**
     *  lista registro da pagina
     */
    private List<TData> data;

    /**
     * Instancia uma pagina de registros
     * @param total total de registro presentes na base de dados
     * @param data lista registro da pagina
     */
    public PageResponse(Long total, List<TData> data) {
        this.total = total;
        this.data = data;
    }
}

File: SpMercantil/Application/Controller/Market/Dto/Response/PageResponse.cs
using System.Collections.Generic;

namespace Application.Controller.Market.Dto.Response
{
    /// <summary>
    ///     Reposta Paginada dos recursos
    /// </summary>
    /// <typeparam name="TData">Tipo do Dado a ser representado na lista</typeparam>
    public class PageResponse<TData>
    {
        /// <summary>
        ///     total de registro presentes na base de dados
        /// </summary>
        public long Total { get; set; }

        /// <summary>
        ///     lista registro da pagina
        /// </summary>
        public List<TData> Data { get; set; }
    }
}